docs/DevelopmentGuide.md

```diff
@@ -11,8 +11,18 @@ Coding Conventions
 * **DO** use `const` and `static` and visibility modifiers to scope exposure of
    variables and methods as much as possible.
 
+* **DO** use doxygen comments, with \[in,out\]
+  [direction annotation](http://www.doxygen.nl/manual/commands.html#cmdparam) in all public API
```

Comment:
It seems like we might want to be more prescriptive about its use in more than just this context of in/out. Maybe we should link out to docs/refman/doxygen-howto.md instead of the general doxygen docs and expand that doc to be more prescriptive.